src/main/flight/mixer.c

```diff
@@ -363,6 +365,15 @@ void mixerInit(mixerMode_e mixerMode)
     idleThrottleOffset = motorConfig()->digitalIdleOffsetValue * 0.0001f;
     idleP = currentPidProfile->idle_p * 0.0001f;
 #endif
+
+    vbatSagCompensationFactor = 0.0f;
+    if (currentPidProfile->vbat_sag_compensation > 0) {
+        vbatFull = batteryConfig()->vbatmaxcellvoltage - 10;
+        vbatRangeToCompensate = vbatFull - batteryConfig()->vbatwarningcellvoltage;
+        if (vbatRangeToCompensate >= 0) {
+            vbatSagCompensationFactor = ((float)currentPidProfile->vbat_sag_compensation) / 100.0f;
+        }
+    }
```

Comment:
`mixerInit()` is only called during flight controller initialization. So the values will stay set to the initial profile settings throughout the entire battery. If the user changes profile (or auto cell profile does) then the new values will not be operational and it will continue to use the values from power cycle.

Reply:
Once #9608 is merged you can rebase and then move your variable initialization into `mixerInitProfile()` to resolve this problem.

Reply:
Yes, I'd wondered about this myself.  Thanks very much for devising a robust fix.